smartsim/entity/model.py

```diff
@@ -198,8 +200,11 @@ def colocate_db_uds(
         :type socket_permissions: int, optional
         :param db_cpus: number of cpus to use for orchestrator, defaults to 1
         :type db_cpus: int, optional
-        :param limit_app_cpus: whether to limit the number of cpus used by the app, defaults to True
-        :type limit_app_cpus: bool, optional
+        :param limit_db_cpus: whether to limit the number of cpus used by the database defaults to True
+        :type limit_db_cpus: bool, optional
```

Comment:
Similar question here: Is this parameter strictly needed? Can we assume that if `db_cpu_list is not None` that a user wants to limit db cpus to that list

Reply:
Just realized this is a bit vague. To be clear can we do something like this:
```py
    def colocate_db_uds(
        self,
        unix_socket: str = "/tmp/redis.socket",
        socket_permissions: int = 755,
        db_cpus: int = 1,
        limit_db_cpus: bool | str = True,
        # NOTE: no `db_cpu_list` param
        debug: bool = False,
        **kwargs: t.Any,
    ) -> None:
        ...
        if isinstance(limit_db_cpus, bool):
            if limit_db_cpus:
                db_cpu_list = "0" if db_cpus == 1 else f"0-{db_cpus-1}"
            else: 
                db_cpu_list = None
        else:
            db_cpu_list = limit_db_cpus
        # type(db_cpu_list) is now str | None
        ...
        common_options = {
            "cpus": db_cpus,
            # NOTE: no `limit_db_cpus` key
            "db_cpu_list": db_cpu_list,
            "debug": debug,
        }
        ...
```
That way we do not have to pass both parameters deeper into the call stack, when in reality we can infer all of the info we need from `db_cpu_list`